File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/03-Consecutive-Characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        '''
        T: O(n) and S: O(1)
        '''
        if len(s) == 1: return 1
        
        power = 0
        i, j = 0, 0
        while j < len(s):
            if s[i] == s[j]:
                j += 1
            else:
                i += 1
            power = max(power, j - i)
        
        return power

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-11/03-Consecutive-Characters.java
class Solution {
    public int maxPower(String s) {
        // T: O(n) and S: O(1)
        
        int n = s.length();
        if (n == 1) return 1;
        
        int power = 0;
        int i = 0;
        int j = 0;
        while (j < n) {
            if (s.charAt(i) == s.charAt(j)) {
                j++;
            } else {
                i++;
            }
            power = Math.max(power, j - i);
        }
        return power;
    }
}
